cmd/gomod-zip/zip.go

```diff
@@ -133,6 +133,30 @@ func main() {
 	defer dst.Close()
 	zw := zip.NewWriter(dst)
 
+	isSubmodule := map[string]bool{}
+	for _, zf := range zr.File {
+		if zf.Name == prefix+"go.mod" {
+			continue
+		}
+		dir, file := path.Split(zf.Name)
+		if file == "go.mod" {
+			isSubmodule[dir] = true
+		}
+	}
+
+	inSubmodule := func(name string) bool {
+		for {
+			dir, _ := path.Split(name)
+			if len(dir) <= len(prefix) {
+				return false
+			}
+			if isSubmodule[dir] {
+				return true
+			}
+			name = dir[:len(dir)-1]
```

Comment:
`name = strings.TrimSuffix(dir, "/")` might be simpler to read

Reply:
Done